### PR TITLE
Feat/clip annotation

### PR DESCRIPTION
**Pull request by HonzaCuhel**
This PR includes the following changes:

- adding CLIP annotator for classification

### CODE REVIEW

File: datadreamer/pipelines/generate_dataset_from_scratch.py [hunk 1]
```diff
@@ -104,11 +105,18 @@ def parse_args():
         help="Image generator to use",
     )
     parser.add_argument(
-        "--image_annotator",
```
Comment (sokovninn) [inline datadreamer/pipelines/generate_dataset_from_scratch.py:107]: Shouldn't we create a deprecation warning for the arguments that are no longer used?

Comment (HonzaCuhel) [reply datadreamer/pipelines/generate_dataset_from_scratch.py:107]: Yes, we should. Thanks!

Or, if you have another idea for the annotator flag names, I'm open to them. @sokovninn

**Comment by sokovninn** [comment]
Have you compared `clip-vit-large-patch14` and `clip-vit-base-patch32`?

File: datadreamer/pipelines/generate_dataset_from_scratch.py (hunk 1 above)
Comment (sokovninn) [reply datadreamer/pipelines/generate_dataset_from_scratch.py:107]: Ok, an alternative option would be to keep the current name `--image_annotator` and check if `--task` is compatible with the specified annotator, which we should do in any case. But not sure which is better.

**Comment by HonzaCuhel** [comment]
No, I haven't, but I can test it.

File: datadreamer/pipelines/generate_dataset_from_scratch.py (hunk 1 above)
Comment (HonzaCuhel) [reply datadreamer/pipelines/generate_dataset_from_scratch.py:107]: I changed it back to using `--image_annotator` flag and added the checks.

**Comment by HonzaCuhel** [comment]
I've manually compared clip-vit-large-patch14 and clip-vit-base-patch32 annotation on 100 generated images on JP using L4 GPU, I've used batch size 8 for annotation and these are the results:

#### Latency:

- `clip-vit-large-patch14` : 3s
- `clip-vit-base-patch32` : 1s

#### Comparison:

- in 12 cases I found `clip-vit-large-patch14`'s annotation better
- in 7 case I found `clip-vit-base-patch32`'s annotation better

#### Conclusion

Switched to using `clip-vit-base-patch32` for the speed as the performance gap isn't big.

**Review by sokovninn — COMMENTED** [review]
Thanks, looks good! I agree with switching to CLIP base.

**Pull request merged**